src/modules/loadPage.ts

```diff
@@ -33,41 +29,10 @@ const loadPage = function(this: Swup, data: TransitionOptions, popstate: PopStat
 	this.currentPageUrl = getCurrentUrl();
 
 	// Load page data
-	if (this.cache.exists(url)) {
-		// Found in Cache, resolve directly
-		xhrPromise = Promise.resolve(this.cache.getPage(url));
-		this.triggerEvent('pageRetrievedFromCache');
-	} else if (this.preloadPromise && this.preloadPromise.route === url) {
-		// Alreay preloading, re-use
-		xhrPromise = this.preloadPromise;
-		this.preloadPromise = null;
-	} else {
-		// Fetch from server
-		xhrPromise = new Promise((resolve, reject) => {
-			fetch({ ...data, headers: this.options.requestHeaders }, (response) => {
-				if (response.status === 500) {
-					this.triggerEvent('serverError');
-					reject(url);
-					return;
-				}
-				// get json data
-				const page = this.getPageData(response);
-				if (!page || !page.blocks.length) {
-					reject(url);
-					return;
-				}
-				// render page
-				const cacheablePageData = { ...page, url };
-				this.cache.cacheUrl(cacheablePageData);
-				this.triggerEvent('pageLoaded');
-				resolve(cacheablePageData);
-			});
-		});
-	}
+	const fetchPromise = this.fetchPage(data);
```

Comment:
Wouldn't we need a way to take over an already-running fetch promise here? `swup. getPageFetchPromise(url)` seemed like a very good idea to me, where preload plugin could add fetch promises to an object `swup.fetchPromises` and swup could pick those up on demand.